test/differentials/thunks.jl

```diff
@@ -89,4 +91,95 @@
             @test was_unthunked == 1
         end
     end
+
+    @testset "basic math" begin
+        @test 1 == - @thunk(-1)
+        @test 1 == @thunk(2) - 1
+        @test 1 == 2 - @thunk(1)
+        @test 1.0 == @thunk(1) / 1.0
+        @test 1.0 == 1.0 / @thunk(1)
+
+        @test 1 == real(@thunk(1 + 1im))
+        @test 1 == imag(@thunk(1 + 1im))
+        @test 1 + 1im == Complex(@thunk(1 + 1im))
+        @test 1 + 1im == Complex(@thunk(1), @thunk(1))
+    end
+
+    @testset "Base functions" begin
+        v = [1, 2, 3]
+        t = @thunk(v)
+
+        @test eltype(v) === eltype(t)
```

Comment:
that is so weird, i had to change it from `Int64 === eltype(t)` since it appears on Ubuntu `eltype([1, 2, 3]) === Int32`?